OF-2614 - Always allow plugin.xml in be the root of a plugin source tree

Fishbowler opened the pull request:
* When creating a plugin project using src/java for the source root, plugin.xml must be in the root.
* When creating a plugin project using the default src/main/java structure, the plugin.xml must be in the src folder.

This is because the openfire-plugin-assembly specifically looks for this file two directories above.

This makes the building more flexible, searching in the project root for the plugin.xml and html files, but keeps the old rule too to avoid breaking existing plugins.

Most of our plugins (e.g. Rest API, Monitoring, Hazelcast) use `src/java` and so have plugin.xml in the root. There are some exceptions (Push Notification, Thread Dump, MUC RTBL) which use the default structure and store the XML in the `src` directory. Real outliers include the Password Reset plugin (which uses gradle to orchestrate a correct build) and ofmeet (which is a monorepo).

By keeping the original rule, I don't think there's any plugin that would be affected. The only unintended consequence I can imagine is something like this:

```
openfire-mucrtbl-plugin
├── DONT-INCLUDE-MY-SECRETS.xml
├── LICENSE
├── pom.xml
├── README.md
└── src
    ├── changelog.html
    ├── logo_large.png
    ├── logo_small.png
    ├── main
    │   ├── java
    │   │   └── org
    │   │       └── igniterealtime
    │   │           └── openfire
    │   │               └── plugin
    │   │                   └── myplugin
    │   │                       ├── MyPlugin.java
    │   │                       └── MyPluginLib.java
    │   └── web
    │       ├── mypage.jsp
    │       └── WEB-INF
    │           └── web.xml
    ├── plugin.xml
    └── readme.html
```

Here, you'd get the plugin.xml as before, but also the DONT-INCLUDE-MY-SECRETS.xml that previously would have been missed.

Feels unlikely?

guusdk commented:
I don't share your concern about `supersecret.xml` being possibly exposed by this change. The fileset includes a very specific set of files:

- files named `plugin.xml`
- files matching `*.html`

I don't see significant risk in exposing HTML files.

guusdk commented on plugins/openfire-plugin-assembly-descriptor/src/main/resources/assemblies/openfire-plugin-assembly.xml:
```diff
@@ -20,6 +20,15 @@
+                <include>plugin.xml</include>
+            </includes>
+        </fileSet>
+        <fileSet> <!-- Cope with legacy & unusual project structures -->
```
For the record, I mean: add more comments to that file - not: create a new documentation page.

Fishbowler commented:
Rebased, added docs.
Tested against s2stest plugin @ [3577696](https://github.com/Fishbowler/openfire-s2stest-plugin/commit/357769630316e677beed7b5101e09df612555ea0)
